### ISSUE TITLE
Review Data Model (Protos)

### ISSUE THREAD

**Issue by tedsuo** [bogdandrutu/openconsensus#147]
Everyone, please review the current protobuf and metadata definitions and leave comments here.

**Comment by tedsuo** [comment]
### Current Protos:
https://github.com/bogdandrutu/openconsensus/tree/master/sdk/src/main/proto

### Related Metadata Discussions:
https://github.com/bogdandrutu/openconsensus/issues?q=metadata

**Comment by bg451** [comment]
I chatted with @iredelmeier about the model and a concern about using protobuf came up.

If a language does not support protobuf or a user doesn't want to include the protobuf library due to size (e.g. browser javascript), there needs to be a common way to serialize spans and metrics using something like JSON. Clear mappings to JSON exist for most protobuf types, however, the protobuf definitions in the SDK heavily uses `oneof` which do not have defined mappings to JSON. See [the protobuf JSON reference](https://developers.google.com/protocol-buffers/docs/proto3#json) for what is and isn't defined. I think this is something that should be discussed when the protos are moved to their own repo.

**Comment by yurishkuro** [comment]
@bg451 good point, and it's not just JSON, in some languages mapping oneof to types is exceptionally ugly. In Jaeger protos we decided to live with minor possible ambiguities of the data rather than have to deal with oneof.

**Comment by bogdandrutu** [comment]
I think the oneof implementation is as simple as in the JSON you enforce to send only one of these fields. I may be wrong but I can ask.
